ystia/forge
# Add a docker image loader TOSCA component

# Pull Request description

## Description of the change

Added new TOSCA component to be able to load a docker image from a tar archive, as described in [GH-140](https://github.com/ystia/forge/issues/140)
Misc.: fixed some installation issues

### What I did

`org/ystia/docker/images/types.yml`:
Added a TOSCA compoent type `org.ystia.docker.images.ArchiveLoader` to load a docker image from a tar archive
Added as well a TOSCA component type `org.ystia.docker.images.RuntimePathArchiveLoader` to load a docker image from a tar archive whose path is only known at runtime.

`org/ystia/docker/images/playbooks/create.yaml`
`org/ystia/docker/images/playbooks/load_archive.yaml`
`org/ystia/docker/images/playbooks/remove_image.yaml`

Added implementations for these component interfaces

#### Misc - Fixed some installations issues:

`org/ystia/docker/ansible/playbooks/create.yaml`
Updated the repository url to support different versions

`org/ystia/docker/containers/generic/playbooks/create.yaml`
Fixed some pip installation issues

`org/ystia/docker/containers/generic/playbooks/delete.yaml`
`org/ystia/docker/containers/generic/playbooks/start.yaml`
`org/ystia/docker/containers/generic/playbooks/stop.yaml`
Fixed an issue where the fact that the container id can change at startup was not taken into account


`org/ystia/experimental/consul/linux/ansible/playbooks/consul_install.yaml`
Fixed a pip installation issue

### How to verify it

Tested in LEXIS project, where a tar gz is stored in a Distributed Data Infrastructure.
This tar gz is downloaded from the Distributed Infrastructure on a Compute Instance, the path on the compute instance being computed at runtime.
A TOSCA component of type `org.ystia.docker.images.RuntimePathArchiveLoader`is then used to load the docker image from this tar gz.

See corresponding topology template and component instantiation at https://github.com/lexis-project/application-templates/blob/v0.1.3/weather-climate/applications/adms/adms_template.yaml#L585-L600

### Description for the changelog

Add a TOSCA component to load a docker image from a tar archive ([GH-140](https://github.com/ystia/forge/issues/140))

## Applicable Issues

Fixes #140
